presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergOrcFileWriter.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.slice.Slice;
+import io.prestosql.orc.OrcDataSink;
+import io.prestosql.orc.OrcDataSource;
+import io.prestosql.orc.OrcWriteValidation;
+import io.prestosql.orc.OrcWriterOptions;
+import io.prestosql.orc.OrcWriterStats;
+import io.prestosql.orc.metadata.ColumnMetadata;
+import io.prestosql.orc.metadata.CompressionKind;
+import io.prestosql.orc.metadata.OrcColumnId;
+import io.prestosql.orc.metadata.OrcType;
+import io.prestosql.orc.metadata.statistics.ColumnStatistics;
+import io.prestosql.orc.metadata.statistics.DateStatistics;
+import io.prestosql.orc.metadata.statistics.DecimalStatistics;
+import io.prestosql.orc.metadata.statistics.DoubleStatistics;
+import io.prestosql.orc.metadata.statistics.IntegerStatistics;
+import io.prestosql.orc.metadata.statistics.StringStatistics;
+import io.prestosql.plugin.hive.orc.OrcFileWriter;
+import io.prestosql.spi.type.Type;
+import org.apache.iceberg.Metrics;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Types;
+import org.joda.time.DateTimeZone;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+import static io.prestosql.orc.metadata.OrcColumnId.ROOT_COLUMN;
+import static io.prestosql.plugin.iceberg.TypeConverter.ORC_ICEBERG_ID_KEY;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class IcebergOrcFileWriter
+        extends OrcFileWriter
+        implements IcebergFileWriter
+{
+    private final Schema icebergSchema;
+    private final ColumnMetadata<OrcType> orcColumns;
+
+    public IcebergOrcFileWriter(
+            OrcDataSink orcDataSink,
+            Callable<Void> rollbackAction,
+            Schema icebergSchema,
```

Comment:
Make this the first or last argument so that the argument ordering matches the parent class (keeps things simpler when reading the code).